Sources/Host.swift

```diff
@@ -1,9 +1,15 @@
 public protocol Host {
-    func accept(timingOut deadline: Double) throws -> Stream
+    associatedtype StreamType: Stream
+    func accept(timingOut deadline: Double) throws -> StreamType
 }
 
 extension Host {
-    func accept() throws -> Stream {
+    func accept() throws -> StreamType {
         return try accept(timingOut: .never)
     }
 }
+
+public protocol BinaryHost: Host {
+    associatedtype BinaryStreamType: BinaryStream
+    func accept(timingOut deadline: Double) throws -> BinaryStreamType
```

Comment:
I'd prefer `duration` over `deadline`. `deadline` sounds like a timestamp when it's really an interval.